src/app/config/no-config/no-config.component.ts

```diff
@@ -137,9 +152,28 @@ export class NoConfigComponent implements OnInit {
 
   public ngOnInit(): void {
     this.changeProgress();
+
+    const mdns = this._electronService.remote.require('mdns');
+    const browser = mdns.createBrowser(mdns.tcp('octoprint'));
+    browser.on('serviceUp', service => {
+      var node = {
+        'display': service.name.match(/"([^"]+)"/)[1] + ' (' + service.txtRecord.version + ')',
+        'name': service.name.match(/"([^"]+)"/)[1],
+        'version': service.txtRecord.version,
+        'url': service.host.replace(/\.$/, '') + ":" + service.port + service.txtRecord.path.replace(/\/$/, '') + "/api/",
+        // Compare version to make sure it meets the requirement
+        'disable': false
+      };
+
+      this.octoprintNodes[service.host.replace(/\.$/, '').replace('.', '_')] = node;
+    });
+    browser.on('serviceDown', service => {
+      delete this.octoprintNodes[service.host.replace(/\.$/, '').replace('.', '_')];
+    });
+    browser.start();
```

Comment:
Can we put this in `main.js`, this would remove the ngx-electron dependency and the icp links are already there, plus we would get around having to add node to the main angular app.

Reply:
Let me know if you want me to do this.

Reply:
Yeah that is beyond me, this took me long enough to figure out ;)

Reply:
Ok I'll take that one :)